File: backend/src/main/java/com/apocalypse/thefall/config/GameInitializer.java
package com.apocalypse.thefall.config;

import com.apocalypse.thefall.service.MapService;
import jakarta.annotation.PostConstruct;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Component;

@Component
@RequiredArgsConstructor
public class GameInitializer {

    private final MapService mapService;

    @PostConstruct
    public void init() {
        // Crée une carte par défaut si aucune n'existe
        mapService.getOrCreateMap();
    }
}

File: backend/src/main/java/com/apocalypse/thefall/config/GameInitializer.kt
package com.apocalypse.thefall.config;

import com.apocalypse.thefall.service.GenerateMapService;
import jakarta.annotation.PostConstruct;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Component;

@Component
@RequiredArgsConstructor
public class GameInitializer {

    private final GenerateMapService generateMapService;

    @PostConstruct
    public void init() {
        // Crée une carte par défaut si aucune n'existe
        generateMapService.getOrCreateMap();
    }
}